packages/@lwc/engine-core/src/framework/template.ts

```diff
@@ -113,6 +115,38 @@ function validateLightDomTemplate(template: Template, vm: VM) {
     }
 }
 
+export function parseFragment(strings: string[], ...keys: number[]): () => Element {
+    return function (): Element {
+        const {
+            context: { hasScopedStyles, stylesheetToken },
+            shadowMode,
+        } = getVMBeingRendered()!;
+        const classToken = hasScopedStyles && stylesheetToken ? ' ' + stylesheetToken : '';
+        const classAttrToken =
+            hasScopedStyles && stylesheetToken ? ` class="${stylesheetToken}"` : '';
+        const attrToken =
+            stylesheetToken && shadowMode === ShadowMode.Synthetic ? ' ' + stylesheetToken : '';
```

Comment:
```suggestion
        const classToken = hasScopedStyles && !isUndefined(stylesheetToken) ? ' ' + stylesheetToken : '';
        const classAttrToken =
            hasScopedStyles && !isUndefined(stylesheetToken) ? ` class="${stylesheetToken}"` : '';
        const attrToken =
            !isUndefined(stylesheetToken) && shadowMode === ShadowMode.Synthetic ? ' ' + stylesheetToken : '';
```

This probably doesn't make a perf difference, but it would be consistent with what we do elsewhere.